apps/dotcom/client/src/tla/components/TlaFileShareMenu/TlaFileShareMenuSharePage.tsx

```diff
@@ -0,0 +1,192 @@
+	if (!auth) throw Error('should have auth')
+
+	const { userId } = auth
+
+	const handleToggleShared = useCallback(() => {
+		// todo: if there are other users connected to the project, warn that they'll be removed from the project until the project is shared again
+		app.toggleFileShared(userId, fileId)
+	}, [app, userId, fileId])
+
+	return (
+		<TlaShareMenuControl>
+			<TlaShareMenuControlLabel>Share this project</TlaShareMenuControlLabel>
+			<TlaSwitch checked={!!shared} onChange={handleToggleShared} />
+		</TlaShareMenuControl>
+	)
+}
+
+function TlaSelectSharedLinkType({ fileId }: { fileId: TldrawAppFileId }) {
+	const app = useApp()
+	const auth = useAuth()
+	if (!auth) throw Error('should have auth')
+	const { userId } = auth
+
+	const sharedLinkType = useValue(
+		'file',
+		() => {
+			const file = app.store.get(fileId)
+			if (!file) throw Error('could not get that file')
+			return file.sharedLinkType
+		},
+		[app, fileId]
+	)
+
+	const handleSelectChange = useCallback(() => {
+		app.toggleFileShareLinkType(userId, fileId)
+	}, [app, userId, fileId])
+
+	return (
+		<TlaShareMenuControl>
+			<TlaShareMenuControlLabel>Anyone with the link can...</TlaShareMenuControlLabel>
+			<TlaSelect
+				label={sharedLinkType === 'edit' ? 'Edit' : 'View'}
+				value={sharedLinkType}
+				onChange={handleSelectChange}
+			>
+				<option value="edit">Edit</option>
+				<option value="view">View</option>
+			</TlaSelect>
+		</TlaShareMenuControl>
+	)
+}
+
+function TlaCopyLinkButton({ shared, fileId }: { shared: boolean; fileId: TldrawAppFileId }) {
+	const app = useApp()
+
+	const handleCopyLinkClick = useCallback(() => {
+		const { auth } = app.getSessionState()
+		if (!auth) throw Error('should have auth')
+		const { userId } = auth
+
+		// Share the file if it isn't shared already
+		if (!shared) app.toggleFileShared(userId, fileId)
+		// Copy the file URL to clipboard
+		const url = getShareableFileUrl(fileId)
+		copyTextToClipboard(url)
+	}, [app, fileId, shared])
+
+	return <TlaShareMenuCopyButton onClick={handleCopyLinkClick}>Copy link</TlaShareMenuCopyButton>
+}
+
+function TlaCopySnapshotLinkButton({ fileId }: { fileId: TldrawAppFileId }) {
+	const app = useApp()
+
+	const handleCopyLinkClick = useCallback(() => {
+		const { auth } = app.getSessionState()
+		if (!auth) throw Error('should have auth')
+		const { userId } = auth
+
+		// todo: implement snapshot link
+		app.createSnapshotLink(userId, fileId)
+		// Copy the snapshot url to clipboard
+		const url = getSnapshotFileUrl(fileId)
+		copyTextToClipboard(url)
+	}, [app, fileId])
+
+	return (
+		<TlaShareMenuCopyButton onClick={handleCopyLinkClick} type="secondary">
+			Copy snapshot link
+		</TlaShareMenuCopyButton>
+	)
+}
+
+function QrCode({ fileId }: { fileId: TldrawAppFileId }) {
+	const ref = useRef<HTMLImageElement>(null)
+
+	// Save the QR codes in local storage
+	const [qrCode, setQrCode] = useLocalStorageState<string | null>(fileId + 'qr-code-11', null)
```

Comment:
what is the `code-11` for?

Reply:
just lazy bumps of the local storage key